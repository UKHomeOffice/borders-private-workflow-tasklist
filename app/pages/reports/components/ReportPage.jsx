import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import { Link, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import AppConstants from '../../../common/AppConstants';
import HTMLReport from './HTMLReport';
import PowerBIReport from './PowerBIReport';

export const ReportPage = ({ appConfig, location }) => {
  if (!location.state) return <Redirect to={AppConstants.REPORTS_PATH} />;
  const { reportServiceUrl } = appConfig;
  const {
    accessToken,
    embedUrl,
    htmlName,
    id,
    name,
    reportType,
  } = location.state;
  document.title = `${name} | ${AppConstants.APP_NAME}`;

  return (
    <div>
      <Link
        id="backToReports"
        className="govuk-link govuk-back-link govuk-!-font-size-19"
        to="/reports"
      >
        Back to reports
      </Link>

      <div
        style={{
          display: 'flex',
          position: 'relative',
          margin: 'auto',
          justifyContent: 'center',
          height: '100vh',
        }}
      >
        {reportType === 'PowerBIReport' ? (
          <PowerBIReport {...{ accessToken, embedUrl, id, name }} />
        ) : (
          <HTMLReport {...{ htmlName, reportServiceUrl }} />
        )}
      </div>
    </div>
  );
};

ReportPage.propTypes = {
  appConfig: PropTypes.shape({
    reportServiceUrl: PropTypes.string.isRequired,
  }).isRequired,
  location: PropTypes.shape({
    state: PropTypes.shape({
      accessToken: PropTypes.string,
      embedUrl: PropTypes.string,
      htmlName: PropTypes.string,
      id: PropTypes.string,
      name: PropTypes.string.isRequired,
      reportType: PropTypes.string,
    }),
  }).isRequired,
};

export default withRouter(
  connect(
    state => ({
      appConfig: state.appConfig,
    }),
    {},
  )(ReportPage),
);
