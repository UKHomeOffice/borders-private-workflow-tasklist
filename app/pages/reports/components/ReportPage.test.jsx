import React from 'react';
import { ReportPage } from './ReportPage';

describe('Report Page', () => {
  it('renders back button', () => {
    const props = {
      appConfig: {
        reportServiceUrl: 'http://localhost:9000',
      },
      location: {
        state: {
          htmlName: 'HTML Report',
          name: 'Report',
        },
      },
    };
    const wrapper = shallow(<ReportPage {...props} />);
    expect(wrapper.find('#backToReports').exists()).toEqual(true);
  });

  it('renders an HTMLReport for HTML reports', () => {
    const props = {
      appConfig: {
        reportServiceUrl: 'http://localhost:9000',
      },
      location: {
        state: {
          htmlName: 'HTML Report',
          name: 'Report',
        },
      },
    };
    const wrapper = shallow(<ReportPage {...props} />);
    expect(wrapper.find('HTMLReport').exists()).toEqual(true);
  });

  it('renders a PowerBIReport for Power BI reports', () => {
    const props = {
      appConfig: {
        reportServiceUrl: 'http://localhost:9000',
      },
      location: {
        state: {
          accessToken: 'xxx',
          embedUrl: 'http://www.example.com',
          id: 'abc',
          name: 'Report',
          reportType: 'PowerBIReport',
        },
      },
    };
    const wrapper = shallow(<ReportPage {...props} />);
    expect(wrapper.find('Connect(PowerBIReport)').exists()).toEqual(true);
  });

  it('renders a redirect if no location.state found', () => {
    const props = {
      appConfig: {
        reportServiceUrl: 'http://localhost:9000',
      },
      location: {},
    };
    const wrapper = shallow(<ReportPage {...props} />);
    expect(wrapper.find('Redirect').exists()).toEqual(true);
  });

  it('matches snapshot', () => {
    const props = {
      appConfig: {
        reportServiceUrl: 'http://localhost:9000',
      },
      location: {},
    };
    const wrapper = shallow(<ReportPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
