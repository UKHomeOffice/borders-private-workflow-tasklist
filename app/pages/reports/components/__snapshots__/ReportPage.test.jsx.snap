// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report Page matches snapshot 1`] = `
<Redirect
  to="/reports"
/>
`;
